### PR TITLE
Sidebar log with newest messages at the top

### PR DESCRIPTION
**Pull request by drbig**
What it says on the tin.

This has been suggested on IRC. Having everything that's important close together is a good idea.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Needs an option to revert it back to current behavior.

**Comment by drbig** [comment]
@Coolthulhu Any tips where I should stick reading the new option so that it gets updated on settings update but I don't need to check the `ACTIVE_WORLD_OPTIONS` on every messages drawing?

**Comment by Coolthulhu** [comment]
uistate.h looks like a good place, I think. Then it would be read from options at the end of options processing, like diag distance setting.

**Comment by Coolthulhu** [comment]
I checked both `trigdist` and `use_tiles`: they're defined in options.cpp, but also declared as `extern bool` in game.h. In this case adding it to game.h would not be a good idea, but adding it as `extern bool` on top of messages.cpp (and actually defining the bool in options.cpp) would make sense.

**Comment by drbig** [comment]
Aye, thanks for the tips. I'm working on this now and should have some commits in today (local time).

**Comment by drbig** [comment]
Looks ok here. I didn't extract the two loops out given we have two options only, and this keeps everything in one place...

File: src/messages.h [hunk 1]
```diff
@@ -13,6 +13,7 @@ class JsonOut;
 class JsonObject;
 
 enum game_message_type : int;
+extern bool log_from_top;
```
Comment (Coolthulhu) [inline src/messages.h]: It shouldn't be in a header when it's only used in one cpp file, but just directly in the cpp file itself.

**Comment by drbig** [comment]
Done.

**Pull request merged**